src/gc/unix/cgroup.cpp

```diff
@@ -0,0 +1,292 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/*++
+
+Module Name:
+
+    cgroup.cpp
+
+Abstract:
+    Read memory limits for the current process
+--*/
+#include <cstdint>
+#include <cstddef>
+#include <cassert>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/resource.h>
+#include <errno.h>
+
+#define MAX_PATH 1024
+#define SIZE_T_MAX (~(size_t)0)
+#define PROC_MOUNTINFO_FILENAME "/proc/self/mountinfo"
+#define PROC_CGROUP_FILENAME "/proc/self/cgroup"
+#define PROC_STATM_FILENAME "/proc/self/statm"
+#define MEM_LIMIT_FILENAME "/memory.limit_in_bytes"
+#define _countof(_array) (sizeof(_array)/sizeof(_array[0]))
+
+class CGroup
+{
+    char m_memory_cgroup_path[MAX_PATH];
+public:
+    CGroup()
+    {
+        if (!FindMemoryHierarchyMount(m_memory_cgroup_path, MAX_PATH))
+        {
+            m_memory_cgroup_path[0] = '\0';
+            return;
+        }
+        char cgroup_path_relative_to_mount[MAX_PATH];
+        if (!FindCGroupPathForMemorySubsystem(cgroup_path_relative_to_mount, MAX_PATH))
+        {
+            m_memory_cgroup_path[0] = '\0';
+            return;
+        }
+        strncat(m_memory_cgroup_path, cgroup_path_relative_to_mount, 
+                MAX_PATH-strlen(m_memory_cgroup_path)-1);
+    }
+    
+    bool GetPhysicalMemoryLimit(size_t *val)
+    {
+        char mem_limit_filename[MAX_PATH];
```

Comment:
Some time ago, we have made a sweeping change over the CoreCLR native code including PAL to get rid of fixed size buffers for file paths unless the path is guaranteed to never exceed that limit. So introducing such a fixed size buffer here makes me nervous. I would prefer using a dynamically allocated buffer at this place. You can use the length of the line from the mount info file that contains the path as an memory size estimate - we would just waste a little memory that's not needed for the path. The FindMemoryHierarchyMount can then return nullptr or the allocated buffer.
I think that similar thing can be done at other places where you use MAX_PATH.